### ISSUE TITLE
GroupBy query cannot use result cache on broker

### ISSUE THREAD

**Issue by hqx871** [apache/druid#11591]
Hi, I want to enable result cache on broker, but I found it does not support group by query. However, more than 50% of our query is group by query.  Hope help.

**Comment by FrankChen021** [comment]
cache does not work for groupBy v2 queries. See #11370

**Comment by abhishekagarwal87** [comment]
I think we can enable result-level caching on broker for groupBy v2. caching is disabled even before result-level caching came to the fore.

**Comment by hqx871** [comment]
I have tried to make GroupByStrategyV2#isCacheable to return true, which will return false on broker before. Then executing some group by case, there is no error and the result is correct as well.

**Comment by hqx871** [comment]
I finally found the cause from issue groupBy v2: Results not fully merged when caching is enabled on the broker #3820. May be we can just disable the segment-level cache on broker and enable result-level cache.

**Comment by hqx871** [comment]
Thanks @abhishekagarwal87 and @FrankChen021. I have success to replay the Results  #3820 [groupBy v2: Results not fully merged when caching is enabled on the broker] by following steps:
1. make GroupByStrategyV2#isQueryCacheable to return true.
2. ingestion the wikiticker-2015-09-12-sampled.json.gz by dynamic partition it into 20 segments with nearly no more than 2000 rows.
3. then take a sql as follow, and I finally got the partialy merged row as {channel = '#es.wikipedia' and namespace in('Main')} 
```
select TIME_FLOOR(__time,'PT1H') "time",
channel,
namespace,
sum(added) sum_added
from wikipedia_dynamic
where __time >='2015-09-12T00:00:00.000Z'
and __time <'2015-09-13T00:00:00.000Z'
GROUP BY TIME_FLOOR(__time,'PT1H'),channel,namespace
limit 10000
```

Finally, as @abhishekagarwal87 said, the result-level cache can be enabled.
